src/profile-logic/marker-timing.js

```diff
@@ -150,24 +154,23 @@ export function getMarkerTiming(
       intervalMarkerTimingsMap.set(markerLineName, markerTimingsForName);
     }
 
-    // Find the first row where the new marker fits.
-    // Since the markers are sorted, look at the last added marker in this row. If
-    // the new marker fits, go ahead and insert it.
-    const foundMarkerTiming = markerTimingsForName.find(
-      (markerTiming) =>
-        markerTiming.end[markerTiming.length - 1] <= marker.start
-    );
+    // Place the marker in the closest row that is empty.
+    for (let i = 0; i < MAX_STACKING_DEPTH; i++) {
+      // Get or create a row for marker timings.
+      let markerTiming = markerTimingsForName[i];
+      if (!markerTiming) {
+        markerTiming = emptyTiming({ instantOnly: false });
+        markerTimingsForName.push(markerTiming);
+      }
 
-    if (foundMarkerTiming) {
-      addCurrentMarkerToMarkerTiming(foundMarkerTiming);
-      continue;
+      // Since the markers are sorted, look at the last added marker in this row.
+      // If the new marker fits, go ahead and insert it.
+      const markerTimingEnd = markerTiming.end[markerTiming.length - 1];
+      if (markerTimingEnd === undefined || markerTimingEnd <= marker.start) {
+        addCurrentMarkerToMarkerTiming(markerTiming);
+        break;
+      }
     }
-
```

Comment:
When we add a new markerTiming track, then you rely that the markerTiming entry is empty, and then that `markerTiming.end[markerTiming.length - 1]` will be `undefined`, but I think it would be good to be more explicit.
How about keeping the exact same logic to before, but simply add a guard after line 164:

```js
if (markerTimingsForName.length >= MAX_STACKING_DEPTH) {
  // There's too many markers around the same time already, let's ignore this marker.
  continue;
}
```

Reply:
Yeah, that's right that we can keep the same logic and add this guard. Initially I came back to using good-old for loop instead of `find` because sometimes I feel like `find` is actually slower :) But I did some testing now and it looks like for this case it actually makes it faster. So reverting this to `find` sounds like a better path forward. Updated the code.